src/solver/conv_direct_naive_conv_bwd.cpp

```diff
@@ -44,15 +44,14 @@ bool ConvDirectNaiveConvBwd::IsApplicable(const ExecutionContext& ctx,
     if(!ConvDirectNaiveConvIsApplicableByKernelType(ctx, problem))
         return false;
 
+    if(!problem.direction.IsBackwardData())
+        return false;
```

Comment:
Existing code has been moved

Reply:
I agree that the current order of checking conditions may not be optimal, but here we need to do a little research and evaluate the expected benefit from this. It's better to do this in a separate pull request